adapters/handlers/rest/handlers_nodes.go

```diff
@@ -0,0 +1,66 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2022 SeMI Technologies B.V. All rights reserved.
+//
+//  CONTACT: hello@semi.technology
+//
+
+package rest
+
+import (
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/semi-technologies/weaviate/adapters/handlers/rest/operations"
+	"github.com/semi-technologies/weaviate/adapters/handlers/rest/operations/nodes"
+	"github.com/semi-technologies/weaviate/adapters/handlers/rest/state"
+	"github.com/semi-technologies/weaviate/adapters/repos/db"
+	"github.com/semi-technologies/weaviate/entities/backup"
+	"github.com/semi-technologies/weaviate/entities/models"
+	"github.com/semi-technologies/weaviate/usecases/auth/authorization/errors"
+	nodesUC "github.com/semi-technologies/weaviate/usecases/nodes"
+	schemaUC "github.com/semi-technologies/weaviate/usecases/schema"
+)
+
+type nodesHandlers struct {
+	manager *nodesUC.Manager
+}
+
+func (s *nodesHandlers) getNodesStatus(params nodes.NodesGetParams, principal *models.Principal) middleware.Responder {
+	nodeStatuses, err := s.manager.GetNodeStatuses(params.HTTPRequest.Context(), principal)
+	if err != nil {
+		switch err.(type) {
+		case errors.Forbidden:
+			return nodes.NewNodesGetForbidden().
+				WithPayload(errPayloadFromSingleErr(err))
+		case backup.ErrUnprocessable:
+			return nodes.NewNodesGetUnprocessableEntity().
+				WithPayload(errPayloadFromSingleErr(err))
+		case backup.ErrNotFound:
```

Comment:
We should probably generalize these errors into the entities package since they are now being used outside of backup-related code